ReactiveCocoa/Swift/EventLogger.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  EventLogger.swift
+//  ReactiveCocoa
+//
+//  Created by Rui Peres on 30/04/2016.
+//  Copyright © 2016 GitHub. All rights reserved.
+//
+
+import Foundation
+
+/// Used with the `debug` operator to customize
+/// the `Event` string.
+///
+/// This is useful, when you want to do more than
+/// just printing to the standard output.
+public protocol EventLogger {
+	func logEvent(event: String, fileName: String, functionName: String, lineNumber: Int)
+}
+
+extension EventLogger {
+	func logEvent(event: String, fileName: String = #file, functionName: String = #function, lineNumber: Int = #line) {
+		print("\(event) fileName: \(fileName), functionaName: \(functionName), lineNumber: \(lineNumber)")
+	}
+}
+
+public enum LoggingEvent: String {
+	case Started, Next, Completed, Failed, Terminated, Disposed, Interrupted
```

Comment:
Would it make sense to make this an inner `enum` on `Signal` and `SignalProducer`? Then `Started` could be removed as a value for `Signal`.